File: src/main/java/com/carolin_violet/travel_system/bean/TouristRoute.java
package com.carolin_violet.travel_system.bean;

import com.baomidou.mybatisplus.annotation.*;

import java.util.Date;
import java.io.Serializable;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.Accessors;

/**
 * <p>
 * 
 * </p>
 *
 * @author carolin-violet
 * @since 2022-05-22
 */
@Data
@EqualsAndHashCode(callSuper = false)
@Accessors(chain = true)
public class TouristRoute implements Serializable {

    private static final long serialVersionUID = 1L;

    /**
     * 线路id
     */
    @TableId(value = "id", type = IdType.ID_WORKER_STR)
    private String id;

    /**
     * 线路标题
     */
    private String title;

    /**
     * 线路描述
     */
    private String description;

    /**
     * 推荐原因
     */
    private String reason;

    private Float price;

    private Float discountPrice;

    /**
     * 逻辑删除
     */
    @TableLogic
    @TableField(fill = FieldFill.INSERT)
    private Integer isDeleted;

    /**
     * 添加时间
     */
    @TableField(fill = FieldFill.INSERT)
    private Date createTime;

    /**
     * 删除时间
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private Date modifyTime;

}

File: server/model/tourist_route.js
const Sequelize = require('sequelize')
const MySequelize = require('../config/mysqlConfig')

const touristRouteModel = MySequelize.define('tourist_route', {
  id: {
    type: Sequelize.STRING(19),
    field:  'id',
    primaryKey: true
  },
  title: {
    type: Sequelize.STRING(25),
    field:  'title',
    allowNull: false
  },
  description: {
    type: Sequelize.STRING(255),
    field:  'description',
    allowNull: false
  },
  reason: {
    type: Sequelize.STRING(255),
    field:  'reason',
    allowNull: false
  },
  price: {
    type: Sequelize.DECIMAL(10,2),
    field:  'price',
    allowNull: false
  },
  discount_price: {
    type: Sequelize.DECIMAL(10,2),
    field:  'discount_price',
    allowNull: false
  },
  is_deleted: {
    type: Sequelize.TINYINT,
    field:  'is_deleted',
    allowNull: false
  },
  create_time: {
    type: Sequelize.DATE,
    field:  'create_time',
    allowNull: false
  },
  modify_time: {
    type: Sequelize.DATE,
    field:  'modify_time',
    allowNull: false
  },
})

module.exports = touristRouteModel